addons/mail/static/src/messaging/component/activity/activity_tests.js

```diff
@@ -780,6 +780,109 @@ QUnit.test('activity click on edit', async function (assert) {
     );
 });
 
+QUnit.test('activity edition', async function (assert) {
+    assert.expect(15);
+    this.data['mail.activity'].records = [{
+        can_write: true,
+        icon: 'fa-check',
+        id: 12,
+        res_id: 42,
+        res_model: 'res.partner',
+    }];
+
+    await this.start({
+        intercepts: {
+            do_action(ev) {
+                assert.step('do_action');
+                assert.strictEqual(
+                    ev.data.action.context.default_res_id,
+                    42,
+                    'Action should have the activity res id as default res id in context'
+                );
+                assert.strictEqual(
+                    ev.data.action.context.default_res_model,
+                    'res.partner',
+                    'Action should have the activity res model as default res model in context'
+                );
+                assert.strictEqual(
+                    ev.data.action.type,
+                    'ir.actions.act_window',
+                    'Action should be of type "ir.actions.act_window"'
+                );
+                assert.strictEqual(
+                    ev.data.action.res_model,
+                    'mail.activity',
+                    'Action should have "mail.activity" as res_model'
+                );
+                assert.strictEqual(
+                    ev.data.action.res_id,
+                    12,
+                    'Action should have activity id as res_id'
+                );
+                ev.data.options.on_close();
+            },
+        },
+        async mockRPC(route, args) {
+            if (route === '/web/dataset/call_kw/mail.activity/activity_format') {
+                assert.step('activity_format');
+            }
+            return this._super(...arguments);
+        },
+    });
+    const activity = this.env.entities.Activity.create({
+        id: 12,
+        can_write: true,
+        icon: 'fa-times',
+        res_id: 42,
+        res_model: 'res.partner',
+    });
+    await this.createActivityComponent(activity);
+
+    assert.containsOnce(
+        document.body,
+        '.o_Activity',
+        "should have activity component"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Activity_editButton',
+        "should have activity edit button"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Activity_icon',
+        "should have activity icon"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Activity_icon .fa-times',
```

Comment:
Hm... weird that `.fa-X` is not on `.o_Activity_icon` itself.
Maybe rename `o_Activity_icon` to `o_Activity_iconContainer`, and add `o_Activity_icon` on the inner `<i/>`.